src/components/node.js

```diff
@@ -87,6 +88,8 @@ class Node extends React.Component {
    */
 
   shouldComponentUpdate = (nextProps) => {
+    if (nextProps.readOnly !== this.props.readOnly) return true
```

Comment:
Ok for the comment.
About the down shift of the row that checks read-only status changes, I think it's not strictly needed.
The value returned by `shouldComponentUpdate` is not affected by the shift.
If `readOnly` property has changed, it is, in any case, `true`.
The same if `readOnly` property hasn't changed but the component rendering the node has `suppressShouldComponentUpdate` set to true.
Finally if `readOnly` property hasn't changed and the component does not have `suppressShouldComponentUpdate` set to true, the returned value depends, in any case, on the rows below.
Anyway, if you prefer I'll move down the row. :blush: